examples/hello_world/android/settings.gradle

```diff
@@ -6,6 +6,27 @@
 // To update all the settings.gradle files in the Flutter repo,
 // See dev/tools/bin/generate_gradle_lockfiles.dart.
 
+pluginManagement {
+    def localProperties = new Properties()
+    def localPropertiesFile = file("local.properties")
+    if (localPropertiesFile.exists()) {
+        localPropertiesFile.withReader("UTF-8") { reader ->
+            localProperties.load(reader)
```

Comment:
It looks like there's some similar logic below. Can those be deduplicated?

Reply:
Yeah, ideally it should be deduplicated. I just don't know where to move it to: the Flutter Gradle plugin can't be used at this point.

Reply:
Can it live at the top of this `settings.gradle` file, but as only one copy? And then use that one both from this `pluginManagement` block, and also for the `apply from: "$flutterSdkPath/packages/flutter_tools/gradle/app_plugin_loader.gradle"` line below.

A further bonus would be if the `localProperties` definition in the template `app/build.gradle` could be removed in favor of reusing this one that was created in `settings.gradle`. But that one is perhaps best kept out of scope of this PR — that's a duplication that already exists, and deduping it would make the change more complicated when this is a PR where the main risk to manage is already that the change may be complicated.

Reply:
> Can it live at the top of this settings.gradle file, but as only one copy?

I tried doing this, but Gradle fails with `pluginManagement block must be defined first`. I tried defining a global function in https://github.com/flutter/flutter/pull/123511/commits/7c4825f3cdd457f33e9cec36e22c4bea15bddb36 but didn't work as well. Will leave as it is for now.

Also, after some debugging it looks like Groovy has some pretty awkward scoping rules. This makes me even more determined to make KTS the default for Gradle buildscripts in Flutter projects :)

Reply:
Ah, Groovy. So be it, then :)

Reply:
According to [this SO answer](https://stackoverflow.com/a/75926948/7009800), this problem would become trivial if `flutter.sdk` was in `gradle.properties` instead of `local.properties`.

Reply:
Resolved 🎉 See [this comment](https://github.com/flutter/flutter/pull/123511#issuecomment-1497932992).